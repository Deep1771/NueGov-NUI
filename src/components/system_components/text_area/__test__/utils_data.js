export let data = {},
  fieldmeta = {
    name: "textarea",
    title: "Textarea",
    type: "TEXTAREA",
    placeHolder: "type here",
    cols: 35,
    rows: 5,
    required: true,
    displayOnCsv: true,
    visible: false,
    disable: true,
    canUpdate: false,
    validationRegEx: /^\w{0,10}$/,
    info: "hi I am a text area",
  };
