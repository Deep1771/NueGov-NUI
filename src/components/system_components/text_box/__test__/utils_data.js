export const fieldmeta = {
  name: "textbox",
  title: "enter value",
  type: "TEXTBOX",
  placeHolder: "select textbox",
  info: "select value",
  visible: false,
  canUpdate: false,
  disable: false,
  required: true,
  validationRegEx:
    /[\w]+((?:[\w-]+\.)*[A-Za-z0-9])+@(?:[A-Za-z0-9-]+\.)+[\w]{2,}/,
  visibleOnCsv: false,
};

export const data = {};
