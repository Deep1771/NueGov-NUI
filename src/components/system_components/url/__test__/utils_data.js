export let data = {},
  fieldmeta = {
    name: "url",
    title: "url",
    type: "URL",
    placeHolder: "Enter url here",
    info: "Enter a url",
    visible: false,
    canUpdate: false,
    disable: false,
    required: false,
    visibleOnCsv: false,
  };
