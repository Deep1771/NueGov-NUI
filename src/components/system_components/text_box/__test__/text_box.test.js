import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  getByPlaceholderText,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import { SystemTextbox } from "../index";
import { data, fieldmeta } from "./utils_data";

afterEach(cleanup);

const onChange = jest.fn();

const defaultProps = {
  data,
  fieldmeta,
  onChange,
};

const setup = () => {
  const utils = render(<SystemTextbox {...defaultProps} />);
  const input = utils.getByPlaceholderText("select textbox");
  const tree = renderer.create(<SystemTextbox {...defaultProps} />).toJSON();

  return {
    input,
    tree,
    ...utils,
  };
};

describe("SystemTextbox Test", () => {
  const { input, tree } = setup();
  test("should contain metadata and data", () => {
    expect(defaultProps.fieldmeta).toBeTruthy();
    expect(defaultProps.data).toBeTruthy();
  });

  test("Check SystemTextbox be required field or not ", () => {
    expect(input).toBeRequired();
  });

  test("check SystemTextbox be disabled or not ", () => {
    expect(input).not.toBeDisabled();
  });

  test("check SystemTextbox email validation", () => {
    fireEvent.change(input, { target: { value: "acewvr@gmail.com" } });
    expect(input.value).toMatch(defaultProps.fieldmeta.validationRegEx);
  });

  test("SystemTextbox changes the text", () => {
    expect(input.value).toMatch("");
    fireEvent.change(input, { target: { value: "enter value" } });
    expect(input.value).toBe("enter value");
  });

  test("snapshot component", () => {
    expect(tree).toMatchSnapshot();
  });
});
