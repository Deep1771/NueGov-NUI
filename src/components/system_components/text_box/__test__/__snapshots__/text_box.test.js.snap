// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemTextbox Test snapshot component 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
  style={
    Object {
      "width": "undefined%",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
    data-shrink={false}
  >
    enter value
    <span
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
      â€‰
      *
    </span>
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="select textbox"
      required={true}
      rows={4}
      type="text"
      value=""
    />
  </div>
</div>
`;
